youtube_dl/extractor/mediaset.py

```diff
@@ -21,7 +21,7 @@ class MediasetIE(ThePlatformBaseIE):
                     (?:
                         mediaset:|
                         https?://
-                            (?:(?:www|static3)\.)?mediasetplay\.mediaset\.it/
+                            (?:(?:www|static3)\.)?mediaset(?:play|infinity)\.mediaset\.it/
```

Comment:
`static3.mediasetinfinity.mediaset.it` doesn't exist.
```suggestion
                            (?:(?:www|(?P<s3>static3))\.)?mediaset(?(s3)play|(?:play|infinity))\.mediaset\.it/
```
Or
```suggestion
                            (?:(?:www\.)?mediasetinfinity|
                               (?:www|static3)\.)?mediasetplay)
                                 \.mediaset\.it/
```